src/emulator/functionsEmulator.ts

```diff
@@ -360,7 +360,7 @@ export class FunctionsEmulator implements EmulatorInstance {
   async stop(): Promise<void> {
     this.workQueue.stop();
     this.workerPool.exit();
-    Promise.resolve(this.server && this.server.close());
+    await Promise.resolve(this.server && this.server.close());
```

Comment:
Why `await` a `resolve()`?  I'd remove `async` and return it to be consistent with other changes.

Reply:
```suggestion
    this.server && this.server.close();
    return Promise.resolve();
```

Reply:
FYI the existing code is simply wrong -- it creates a Promise that nobody consumes.

Reply:
`this.server.close` takes a callback - should that be used in a resolver?